TestUtilitiesInterface/EntityComparatorMacros.h

```diff
@@ -27,3 +27,9 @@ using namespace testing;
 			return AssertionFailure() << "Result for expression " #EXPRESSION " is different: expected=" << EXPECTED.EXPRESSION << ", actual=" << TO_TEST.EXPRESSION;\
 		}\
 	}
+
+#define ENTITY_COMPARATOR_ASSERT_EQUAL(EXPECTED, TO_TEST) \
```

Comment:
This macro can be very useful, but I think that we should add context when the compared entities fail. We are just returning the result than the comparator of the entities, but we should extend the message to indicate the expression / attribute used to gather the compared entities. This way, we will be able to identify the differences when the compared object has multiple instances of the same type.

Reply:
```
#define ENTITY_COMPARATOR_ASSERT_EQUAL(EXPECTED, TO_TEST, EXPRESSION) \
	{ \
		auto result = EntityComparator()(EXPECTED.EXPRESSION, TO_TEST.EXPRESSION); \
		if (!result) \
		{ \
			return AssertionFailure() << "Result for expression " #EXPRESSION " is different:" << result.message(); \
		} \
	}

```

Reply:
Additionally, we should be able to use that when comparing 2 vectors of entities, where we should know the number of iteration that failed the comparison.
```
		unsigned int nItems = expected.count();
		for (unsigned int i = 0; i < nItems ; i++)
		{
			ENTITY_COMPARATOR_ITERATION_ASSERT_EQUAL(expected, toTest, i);
		}
```
Maybe another macro with an integer counter would do the job.